### PR TITLE
jQuery.effects.define()

### PR DESCRIPTION
**Pull request by scottgonzalez**

### CODE REVIEW

**Comment by scottgonzalez** [comment]
I added `jQuery.effects.scaledDimensions()` to this PR.

**Comment by agcolom** [comment]
Looks good :-) (i.e. no typo)

**Comment by scottgonzalez** [comment]
I added `jQuery.effects.clipToBox()` and `jQuery.effects.createPlaceholder()` to this PR.

**Comment by scottgonzalez** [comment]
I added `jQuery.effects.removePlaceholder()`, `jQuery.effects.saveStyle()`, and `jQuery.effects.restoreStyle()` to this PR.

**Comment by agcolom** [comment]
All looks good :-)

File: entries/jQuery.effects.saveStyle.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	<entry type="method" name="jQuery.effects.saveStyle" added="1.12">
+		<title>jQuery.effects.saveStyle()</title>
+		<signature>
+			<argument name="element" type="jQuery">
+				<desc>The element to save styles for.</desc>
+			</argument>
+		</signature>
+		<desc>Saves all inline styles applied to an element.</desc>
+		<longdesc>
+			<p>Saves all inline styles applied to an element. This is useful when animating various styles and restoring the existing styles at the end. The saved styles can be restored using <a href="/jQuery.effects.restoreStyle/"><code>jQuery.effects.restoreStyle()</code></a>.</p>
```
Comment (jzaefferer) [inline entries/jQuery.effects.saveStyle.xml]: This needs to explain what "save" means here. As far as I can tell it stores a copy of `style.cssText` via `.data()`, without making any changes to the styles itself.

Also applies to `restoreStyle`.

**Comment by jzaefferer** [comment]
Apart from the underspecified `saveStyle` everything looks good.

File: entries/jQuery.effects.saveStyle.xml (hunk 1 above)
Comment (scottgonzalez) [inline entries/jQuery.effects.saveStyle.xml]: The how is pure internal details. The only thing the user should care about is how to restore the styles that have been saved, which is documented.

Comment (jzaefferer) [inline entries/jQuery.effects.saveStyle.xml]: Its not just internal details. The semantics of the method were unclear to me until I read the code, which makes for bad documentation.

Comment (scottgonzalez) [inline entries/jQuery.effects.saveStyle.xml]: Can you explain what semantics were unclear?

**Comment by scottgonzalez** [comment]
Updated the wording for `saveStyle()`.

**Comment by jzaefferer** [comment]
Looks good

**Comment by scottgonzalez** [comment]
Merged into 1-12.

**Pull request closed without merging**